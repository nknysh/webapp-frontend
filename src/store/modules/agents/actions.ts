import { ICompany, ITravelAgent } from 'services/BackendApi';
export const GET_TRAVEL_AGENTS_REQUEST = 'agentsModule/GET_TRAVEL_AGENTS_REQUEST';
export const GET_TRAVEL_AGENTS_SUCCESS = 'agentsModule/GET_TRAVEL_AGENTS_SUCCESS';
export const GET_TRAVEL_AGENTS_FAILURE = 'agentsModule/GET_TRAVEL_AGENTS_FAILURE';
export const SELECTED_TA_CHANGE = 'agentsModule/SELECTED_TA_CHANGE';
export const SHOW_TA_DROPDOWN = 'agentsModule/SHOW_TA_DROPDOWN';
export const SEARCH_TA_BY_NAME_CHANGE = 'agentsModule/SEARCH_TA_BY_NAME_CHANGE';
export const GET_COMPANIES_REQUEST = 'agentsModule/GET_COMPANIES_REQUEST';
export const GET_COMPANIES_SUCCESS = 'agentsModule/GET_COMPANIES_SUCCESS';
export const GET_COMPANIES_FAILURE = 'agentsModule/GET_COMPANIES_FAILURE';
export const SELECTED_COMPANY_CHANGE = 'agentsModule/SELECTED_COMPANY_CHANGE';
export const SHOW_COMPANY_DROPDOWN = 'agentsModule/SHOW_COMPANY_DROPDOWN';
export const SEARCH_COMPANY_BY_NAME = 'agentsModule/SEARCH_COMPANY_BY_NAME';

export type GetTravelAgentRequestAction = ReturnType<typeof getTravelAgentsRequestAction>;
export const getTravelAgentsRequestAction = (forceLoad?: boolean) => ({
  type: GET_TRAVEL_AGENTS_REQUEST as typeof GET_TRAVEL_AGENTS_REQUEST,
  forceLoad,
});

export type GetTravelAgentSuccessAction = ReturnType<typeof getTravelAgentsSuccessAction>;
export const getTravelAgentsSuccessAction = (agents: ITravelAgent[]) => ({
  type: GET_TRAVEL_AGENTS_SUCCESS as typeof GET_TRAVEL_AGENTS_SUCCESS,
  agents,
});

export type GetTravelAgentFailureAction = ReturnType<typeof getTravelAgentsFailureAction>;
export const getTravelAgentsFailureAction = (error: string) => ({
  type: GET_TRAVEL_AGENTS_FAILURE as typeof GET_TRAVEL_AGENTS_FAILURE,
  error,
});

export type SelectedTaChangeAction = ReturnType<typeof selectedTaChangeAction>;
export const selectedTaChangeAction = (value: ITravelAgent | null) => ({
  type: SELECTED_TA_CHANGE as typeof SELECTED_TA_CHANGE,
  value,
});

export type ShowTaDropdownAction = ReturnType<typeof showTaDropdownAction>;
export const showTaDropdownAction = (value: boolean) => ({
  type: SHOW_TA_DROPDOWN as typeof SHOW_TA_DROPDOWN,
  value,
});

export type SearchTaByNameChangeAction = ReturnType<typeof searchTaByNameChangeAction>;
export const searchTaByNameChangeAction = (value: string) => ({
  type: SEARCH_TA_BY_NAME_CHANGE as typeof SEARCH_TA_BY_NAME_CHANGE,
  value,
});


export type GetCompaniesRequestAction = ReturnType<typeof getCompaniesRequestAction>;
export const getCompaniesRequestAction = () => ({
  type: GET_COMPANIES_REQUEST as typeof GET_COMPANIES_REQUEST,
});

export type GetCompaniesSuccessAction = ReturnType<typeof getCompaniesSuccessAction>;
export const getCompaniesSuccessAction = (companies: ICompany[]) => ({
  type: GET_COMPANIES_SUCCESS as typeof GET_COMPANIES_SUCCESS,
  companies,
});

export type GetCompaniesFailureAction = ReturnType<typeof getCompaniesFailureAction>;
export const getCompaniesFailureAction = (error: string) => ({
  type: GET_COMPANIES_FAILURE as typeof GET_COMPANIES_FAILURE,
  error,
});

export type SelectedCompanyChangeAction = ReturnType<typeof selectedCompanyChangeAction>;
export const selectedCompanyChangeAction = (value: ICompany) => ({
  type: SELECTED_COMPANY_CHANGE as typeof SELECTED_COMPANY_CHANGE,
  value,
});

export type ShowCompanyDropdownAction = ReturnType<typeof showCompanyDropdownAction>;
export const showCompanyDropdownAction = (value: boolean) => ({
  type: SHOW_COMPANY_DROPDOWN as typeof SHOW_COMPANY_DROPDOWN,
  value,
});

export type SearchCompanyByNameAction = ReturnType<typeof searchCompanyByNameAction>;
export const searchCompanyByNameAction = (value: string) => ({
  type: SEARCH_COMPANY_BY_NAME as typeof SEARCH_COMPANY_BY_NAME,
  value,
});

export type AgentsModuleAction =
  | GetTravelAgentRequestAction
  | GetTravelAgentSuccessAction
  | GetTravelAgentFailureAction
  | SelectedTaChangeAction
  | ShowTaDropdownAction
  | SearchTaByNameChangeAction
  | GetCompaniesRequestAction
  | GetCompaniesSuccessAction
  | GetCompaniesFailureAction
  | SelectedCompanyChangeAction
  | ShowCompanyDropdownAction
  | SearchCompanyByNameAction;
