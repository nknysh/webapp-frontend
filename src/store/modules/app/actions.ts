export const SET_LATEST_APP_VERSION = 'app/SET_LATEST_APP_VERSION';

export type SetLatestAppVersion = ReturnType<typeof setLatestAppVersion>;
export const setLatestAppVersion = (latestAppVersion: string, isAppDeprecated: boolean) => ({
  type: SET_LATEST_APP_VERSION as typeof SET_LATEST_APP_VERSION,
  latestAppVersion,
  isAppDeprecated
});


export type AppAction =
 | SetLatestAppVersion
