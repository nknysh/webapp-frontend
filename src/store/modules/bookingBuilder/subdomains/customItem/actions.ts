export const SHOW_CUSTOM_ITEM_FORM = 'bookingBuilder/customItem/SHOW_CUSTOM_ITEM_FORM';
export const HIDE_CUSTOM_ITEM_FORM = 'bookingBuilder/customItem/HIDE_CUSTOM_ITEM_FORM';

export const UPDATE_CUSTOM_ITEM_NAME = 'bookingBuilder/customItem/UPDATE_CUSTOM_ITEM_NAME';
export const UPDATE_CUSTOM_ITEM_TOTAL = 'bookingBuilder/customItem/UPDATE_CUSTOM_ITEM_TOTAL';
export const UPDATE_CUSTOM_ITEM_DESCRIPTION = 'bookingBuilder/customItem/UPDATE_CUSTOM_ITEM_DESCRIPTION';
export const UPDATE_CUSTOM_ITEM_COUNTS_AS_MEAL_PLAN = 'bookingBuilder/customItem/UPDATE_CUSTOM_ITEM_COUNTS_AS_MEAL_PLAN';
export const UPDATE_CUSTOM_ITEM_COUNTS_AS_TRANSFER = 'bookingBuilder/customItem/UPDATE_CUSTOM_ITEM_COUNTS_AS_MEAL_PLAN';


export type ShowCustomItemFormAction = ReturnType<typeof showCustomItemFormAction>;
export const showCustomItemFormAction = () => ({
  type: SHOW_CUSTOM_ITEM_FORM as typeof SHOW_CUSTOM_ITEM_FORM
});

export type HideCustomItemFormAction = ReturnType<typeof hideCustomItemFormAction>;
export const hideCustomItemFormAction = () => ({
  type: HIDE_CUSTOM_ITEM_FORM as typeof HIDE_CUSTOM_ITEM_FORM
});

//------------------- edit ----------------------------------

export type UpdateCustomItemNameAction = ReturnType<typeof updateCustomItemNameAction>;
export const updateCustomItemNameAction = (value: string) => ({
  type: UPDATE_CUSTOM_ITEM_NAME as typeof UPDATE_CUSTOM_ITEM_NAME,
  value
});

export type UpdateCustomItemTotalAction = ReturnType<typeof updateCustomItemTotalAction>;
export const updateCustomItemTotalAction = (value: string) => ({
  type: UPDATE_CUSTOM_ITEM_TOTAL as typeof UPDATE_CUSTOM_ITEM_TOTAL,
  value
});

export type UpdateCustomItemDescriptionAction = ReturnType<typeof updateCustomItemDescriptionAction>;
export const updateCustomItemDescriptionAction = (value: string) => ({
  type: UPDATE_CUSTOM_ITEM_DESCRIPTION as typeof UPDATE_CUSTOM_ITEM_DESCRIPTION,
  value
});

export type UpdateCustomItemCountsAsMealPlanAction = ReturnType<typeof updateCustomItemCountsAsMealPlanAction>;
export const updateCustomItemCountsAsMealPlanAction = (value: boolean) => ({
  type: UPDATE_CUSTOM_ITEM_COUNTS_AS_MEAL_PLAN as typeof UPDATE_CUSTOM_ITEM_COUNTS_AS_MEAL_PLAN,
  value
});

export type UpdateCustomItemCountsAsTransferAction = ReturnType<typeof updateCustomItemCountsAsTransferAction>;
export const updateCustomItemCountsAsTransferAction = (value: boolean) => ({
  type: UPDATE_CUSTOM_ITEM_COUNTS_AS_TRANSFER as typeof UPDATE_CUSTOM_ITEM_COUNTS_AS_TRANSFER,
  value
});


export type CustomItemAction =
  | ShowCustomItemFormAction
  | HideCustomItemFormAction

  | UpdateCustomItemNameAction
  | UpdateCustomItemTotalAction
  | UpdateCustomItemDescriptionAction
  | UpdateCustomItemCountsAsMealPlanAction
  | UpdateCustomItemCountsAsTransferAction;
