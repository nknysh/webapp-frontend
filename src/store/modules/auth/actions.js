import { prop } from 'ramda';

import { successAction, errorAction } from 'store/common/actions';

import AuthApi from 'api/auth';

export const AUTH_REQUEST = 'AUTH_REQUEST';
export const AUTH_OK = 'AUTH_OK';
export const AUTH_ERROR = 'AUTH_ERROR';
export const AUTH_RESET = 'AUTH_RESET';
export const AUTH_SET_TOKEN = 'AUTH_SET_TOKEN';
export const AUTH_SIGN_UP = 'AUTH_SIGN_UP';

// This constant is for Localstorage.
const AUTH_TOKEN = 'authToken';

const authRequest = value => ({
  type: AUTH_REQUEST,
  payload: value,
});

export const authOk = value => {
  const token = prop('token', value);
  // TODO(mark): We should use something like sagas to store this token.
  // Right now, this is called when we logIn and getUserFromToken.
  if (token) {
    localStorage.setItem(AUTH_TOKEN, token);
  }

  return {
    type: AUTH_OK,
    payload: value,
  };
};

const authError = value => {
  localStorage.removeItem(AUTH_TOKEN);

  return {
    type: AUTH_ERROR,
    payload: value,
  };
};

const authReset = value => ({
  type: AUTH_RESET,
  payload: value,
});

export const getUserFromToken = ({ token }) => dispatch => {
  dispatch(authRequest({ token }));
  return AuthApi.getUserFromToken({ token })
    .then(response => {
      const ok = { token, user: response.data };
      dispatch(authOk(ok));
    })
    .catch(error => {
      dispatch(authError(error));
      throw error;
    });
};

export const logIn = values => async dispatch => {
  dispatch(authRequest(values));
  try {
    const response = await AuthApi.logIn(values);
    const { id, user } = response.data;
    dispatch(authOk({ token: id, user }));
    return user;
  } catch (error) {
    dispatch(authError(error));
    throw error;
  }
};

export const logOut = values => dispatch => {
  localStorage.removeItem(AUTH_TOKEN);
  dispatch(authReset(values));
};

export const resetPassword = values => async dispatch => {
  dispatch(authRequest(values));

  try {
    const response = await AuthApi.resetPassword(values);
    return response;
  } catch (error) {
    dispatch(authError(error));
    throw error;
  }
};

export const setToken = token => ({
  type: AUTH_SET_TOKEN,
  payload: { token },
});

export const authSignUp = values => ({
  type: AUTH_SIGN_UP,
  payload: values,
});

export const signUp = values => dispatch => {
  dispatch(authSignUp(values));

  // This is where APi call would be handled.
  // return AuthApi.signUp(values).then(successAction).catch(errorAction);

  /**
   * @todo Return from API call the correct action and remove this
   */
  return values ? dispatch(successAction(AUTH_SIGN_UP, values)) : dispatch(errorAction(AUTH_SIGN_UP, values));
};
