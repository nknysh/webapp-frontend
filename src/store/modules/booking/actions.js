import { __, pick, lensProp, lensPath, prop, set, view, map, mapObjIndexed, pipe, mergeDeepRight } from 'ramda';
import { eachDay, subDays } from 'date-fns';

import client from 'api/bookings';
import { formatDate } from 'utils';

import { successAction, errorFromResponse } from 'store/common';
import { enqueueNotification } from 'store/modules/ui/actions';
import { getHotelRoom } from 'store/modules/hotels/selectors';

import { getBookingData, getBookingByHotelId, getBookingRoomDatesById } from './selectors';

const accommodationProductsLens = lensProp('accommodationProducts');
const datesLens = lensProp('dates');
const ratesLens = lensPath(['dates', 'rates']);

export const BOOKING_UPDATE = 'BOOKING_UPDATE';
export const BOOKING_SUBMIT = 'BOOKING_SUBMIT';
export const BOOKING_REMOVE = 'BOOKING_REMOVE';

export const updateBookingAction = payload => ({
  type: BOOKING_UPDATE,
  payload,
});

export const submitBookingAction = payload => ({
  type: BOOKING_SUBMIT,
  payload,
});

export const removeBooking = payload => ({
  type: BOOKING_REMOVE,
  payload,
});

export const updateBooking = (id, payload) => (dispatch, getState) => {
  const state = getState();

  dispatch(updateBookingAction(payload));
  dispatch(successAction(BOOKING_UPDATE, mergeDeepRight(getBookingData(state), { [id]: payload })));
};

export const completeBooking = (id, payload) => async (dispatch, getState) => {
  dispatch(updateBooking(id, payload));

  const state = getState();
  const booking = getBookingByHotelId(state, id);

  const finaliseAccommodationProducts = (product, productId) => {
    const hotelRoom = getHotelRoom(state, id, productId);
    const { to, from } = getBookingRoomDatesById(state, id, productId);

    const rates = prop('rates', hotelRoom);

    const days = eachDay(from, subDays(to, 1));
    const ratesForDays = pipe(
      map(formatDate),
      pick(__, rates)
    )(days);

    return pipe(
      set(datesLens, { to, from }),
      set(ratesLens, ratesForDays)
    )(product);
  };

  const finalBooking = pipe(
    set(
      accommodationProductsLens,
      mapObjIndexed(finaliseAccommodationProducts, view(accommodationProductsLens, booking))
    )
  )(booking);

  dispatch(submitBookingAction(finalBooking));

  try {
    const {
      data: { data },
    } = await client.createBooking({ data: { attributes: { id, ...finalBooking } } });

    dispatch(updateBooking(id, { reservationId: prop('uuid', data) }));
    dispatch(successAction(BOOKING_SUBMIT, data));
  } catch (e) {
    dispatch(
      enqueueNotification({ message: 'There was a problem creating your booking.', options: { variant: 'error' } })
    );
    dispatch(errorFromResponse(BOOKING_SUBMIT, e));
    throw e;
  }
};
