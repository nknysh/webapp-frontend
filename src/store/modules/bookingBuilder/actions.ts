import {
  IHotel,
  BookingBuilder,
  BookingBuilderResponse,
  TransferReference,
  GuestAges,
  GroundServiceReference,
  FineReference,
  SupplementReference,
  HotelAccommodationProduct,
  ISearchQuery,
} from 'services/BackendApi/types';

import { CustomItemAction } from './subdomains/customItem/actions';

export const INITIALIZE_BOOKING_BUILDER = 'bookingBuilder/INITIALIZE_BOOKING_BUILDER';
export const INITIALIZE_BOOKING_BUILDER_FAILURE = 'bookingBuilder/INITIALIZE_BOOKING_BUILDER_FAILURE';
export const COPY_BOOKING_BUILDER = 'bookingBuilder/COPY_BOOKING_BUILDER';
export const CREATE_STUB_BOOKING_BUILDER = 'bookingBuilder/CREATE_STUB_BOOKING_BUILDER';
export const CLEAR_BOOKING_BUILDER = 'bookingBuilder/CLEAR_BOOKING_BUILDER';

export const UPDATE_TRANSFER = 'bookingBuilder/UPDATE_TRANSFER';
export const UPDATE_BOOKING_SUCCESS = 'bookingBuilder/UPDATE_BOOKING_SUCCESS';

export const UPDATE_GROUND_SERVICE_ACTION = 'bookingBuilder/UPDATE_GROUND_SERVICE_ACTION';
export const UPDATE_SUPPLEMENT_ACTION = 'bookingBuilder/UPDATE_SUPPLEMENT_ACTION';
export const UPDATE_FINE_ACTION = 'bookingBuilder/UPDATE_FINE_ACTION';
// LODGINGS
export const UPDATE_LODGING_GUEST_AGES_ACTION = 'bookingBuilder/UPDATE_LODGING_GUEST_AGES_ACTION';
export const UPDATE_LODGING_DATES_ACTION = 'bookingBuilder/UPDATE_LODGING_DATES_ACTION';
export const UPDATE_LODGING_MEAL_PLAN_ACTION = 'bookingBuilder/UPDATE_LODGING_MEAL_PLAN_ACTION';
export const REMOVE_LODGING_ACTION = 'bookingBuilder/REMOVE_LODGING_ACTION';
export const UPDATE_LODGING_OCCASIONS_ACTION = 'bookingBuilder/UPDATE_LODGING_OCCASIONS_ACTION';

export const ADD_LODGING_ACTION = 'bookingBuilder/ADD_LODGING_ACTION';

export const FORWARDS_COMPAT_BOOKING_BUILDER_ACTION = 'bookingBuilder/FORWARDS_COMPAT_BOOKING_BUILDER_ACTION';

export const UPDATE_TA_MARGIN_TYPE_ACTION = 'bookingBuilder/UPDATE_TA_MARGIN_TYPE_ACTION';
export const UPDATE_TA_MARGIN_AMOUNT_ACTION = 'bookingBuilder/UPDATE_TA_MARGIN_AMOUNT_ACTION';

export const UPDATE_IS_TA_MARGIN_APPLIED_ACTION = 'bookingBuilder/UPDATE_IS_TA_MARGIN_APPLIED_ACTION';

export const UPDATE_BOOKING_GUEST_INFORMATION_ACTION = 'bookingBuilder/UPDATE_BOOKING_GUEST_INFORMATION_ACTION';
export const CLEAR_BOOKING_BUILDER_UI_STATE = 'bookingBuilder/CLEAR_BOOKING_BUILDER_UI_STATE';

export const UPDATE_TRAVEL_AGENT_USER_ID = 'bookingBuilder/UPDATE_TRAVEL_AGENT_USER_ID';

export const UPDATE_LODGING_REPEAT_GUEST_ACTION = 'bookingBuilder/UPDATE_LODGING_REPEAT_GUEST_ACTION';
export const SAVE_CUSTOM_ITEM = 'bookingBuilder/SAVE_CUSTOM_ITEM';
export const REMOVE_CUSTOM_ITEM = 'bookingBuilder/REMOVE_CUSTOM_ITEM';

export const UPDATE_AGREEE_TO_TERMS = 'bookingBuilder/UPDATE_AGREEE_TO_TERMS';
export const SET_IS_PRISTINE = 'bookingBuilder/SET_IS_PRISTINE';
export const SET_LATEST_BOOKING_OPERATION = 'bookingBuilder/SET_LATEST_BOOKING_OPERATION';

export type InitializeBookingBuilderAction = ReturnType<typeof initializeBookingBuilderAction>;
export const initializeBookingBuilderAction = (hotelUuid: string) => ({
  type: INITIALIZE_BOOKING_BUILDER as typeof INITIALIZE_BOOKING_BUILDER,
  hotelUuid,
});

export type InitializeBookingBuilderFailureAction = ReturnType<typeof initializeBookingBuilderFailureAction>;
export const initializeBookingBuilderFailureAction = () => ({
  type: INITIALIZE_BOOKING_BUILDER_FAILURE as typeof INITIALIZE_BOOKING_BUILDER_FAILURE,
});

export type CopyBookingBuilderAction = ReturnType<typeof copyBookingBuilderAction>;
export const copyBookingBuilderAction = (bookingBuilder: BookingBuilder) => ({
  type: COPY_BOOKING_BUILDER as typeof COPY_BOOKING_BUILDER,
  bookingBuilder,
});

export type CreateStubBookingBuilderAction = ReturnType<typeof createStubBookingBuilderAction>;
export const createStubBookingBuilderAction = (hotel: IHotel) => ({
  type: CREATE_STUB_BOOKING_BUILDER as typeof CREATE_STUB_BOOKING_BUILDER,
  hotel,
});

export type ClearBookingBuilderAction = ReturnType<typeof clearBookingBuilderAction>;
export const clearBookingBuilderAction = () => ({
  type: CLEAR_BOOKING_BUILDER as typeof CLEAR_BOOKING_BUILDER,
});

export type UpdateTransferAction = ReturnType<typeof updateTransferAction>;
export const updateTransferAction = (transfer: TransferReference, hotelUuid: string) => ({
  type: UPDATE_TRANSFER as typeof UPDATE_TRANSFER,
  transfer,
  hotelUuid,
});

export type UpdateGroundServiceAction = ReturnType<typeof updateGroundServiceAction>;
export const updateGroundServiceAction = (groundService: GroundServiceReference, hotelUuid: string) => ({
  type: UPDATE_GROUND_SERVICE_ACTION as typeof UPDATE_GROUND_SERVICE_ACTION,
  groundService,
  hotelUuid,
});

export type UpdateSupplementAction = ReturnType<typeof updateSupplementAction>;
export const updateSupplementAction = (supplement: SupplementReference, hotelUuid: string) => ({
  type: UPDATE_SUPPLEMENT_ACTION as typeof UPDATE_SUPPLEMENT_ACTION,
  supplement,
  hotelUuid,
});

export type UpdateFineAction = ReturnType<typeof updateFineAction>;
export const updateFineAction = (fine: FineReference, hotelUuid: string) => ({
  type: UPDATE_FINE_ACTION as typeof UPDATE_FINE_ACTION,
  fine,
  hotelUuid,
});

export type UpdateBookingSuccessAction = ReturnType<typeof updateBookingSuccessAction>;
export const updateBookingSuccessAction = (response: BookingBuilderResponse, hotelUuid: string) => ({
  type: UPDATE_BOOKING_SUCCESS as typeof UPDATE_BOOKING_SUCCESS,
  response,
  hotelUuid,
});

export type UpdateLodgingGuestAgesAction = ReturnType<typeof updateLodgingGuestAgesAction>;
export const updateLodgingGuestAgesAction = (hotelUuid: string, lodgingIndex: number, guestAges: GuestAges) => ({
  type: UPDATE_LODGING_GUEST_AGES_ACTION as typeof UPDATE_LODGING_GUEST_AGES_ACTION,
  hotelUuid,
  lodgingIndex,
  guestAges,
});

export type UpdateLodgingMealPlanAction = ReturnType<typeof updateLodgingMealPlanAction>;
export const updateLodgingMealPlanAction = (hotelUuid: string, lodgingIndex: number, mealPlanUuids: string[]) => ({
  type: UPDATE_LODGING_MEAL_PLAN_ACTION as typeof UPDATE_LODGING_MEAL_PLAN_ACTION,
  hotelUuid,
  lodgingIndex,
  mealPlanUuids,
});

export type RemoveLodgingAction = ReturnType<typeof removeLodgingAction>;
export const removeLodgingAction = (hotelUuid: string, lodgingIndex: number) => ({
  type: REMOVE_LODGING_ACTION as typeof REMOVE_LODGING_ACTION,
  hotelUuid,
  lodgingIndex,
});

export type UpdateLodgingOccasionsAction = ReturnType<typeof updateLodgingOccasionsAction>;
export const updateLodgingOccasionsAction = (hotelUuid: string, lodgingIndex: number, occasions: object) => ({
  type: UPDATE_LODGING_OCCASIONS_ACTION as typeof UPDATE_LODGING_OCCASIONS_ACTION,
  hotelUuid,
  lodgingIndex,
  occasions,
});

export type UpdateLodgingDatesAction = ReturnType<typeof updateLodgingDatesAction>;
export const updateLodgingDatesAction = (
  hotelUuid: string,
  lodgingIndex: number,
  startDate: string,
  endDate: string
) => ({
  type: UPDATE_LODGING_DATES_ACTION as typeof UPDATE_LODGING_DATES_ACTION,
  hotelUuid,
  lodgingIndex,
  startDate,
  endDate,
});

// add lodgings
export type AddLodgingAction = ReturnType<typeof addLodgingAction>;
export const addLodgingAction = (
  accommodationProduct: any, // the accommodation product comes from JS land and is a massive nightmare
  searchQuery: ISearchQuery,
  guestAges?: {
    numberOfAdults: number;
    agesOfAllChildren: number[];
  }
) => ({
  type: ADD_LODGING_ACTION as typeof ADD_LODGING_ACTION,
  accommodationProduct,
  searchQuery,
  guestAges,
});

export type ForwardsCompatBookingBuilderAction = ReturnType<typeof forwardsCompatBookingBuilderAction>;
export const forwardsCompatBookingBuilderAction = (booking: any, holds: any) => ({
  type: FORWARDS_COMPAT_BOOKING_BUILDER_ACTION as typeof FORWARDS_COMPAT_BOOKING_BUILDER_ACTION,
  booking,
  holds,
});

export type UpdateTAMarginType = ReturnType<typeof updateTAMarginTypeAction>;
export const updateTAMarginTypeAction = (hotelUuid: string, taMarginType: string) => ({
  type: UPDATE_TA_MARGIN_TYPE_ACTION as typeof UPDATE_TA_MARGIN_TYPE_ACTION,
  hotelUuid,
  taMarginType,
});

export type UpdateTAMarginAmount = ReturnType<typeof updateTAMarginAmountAction>;
export const updateTAMarginAmountAction = (hotelUuid: string, taMarginAmount: string) => ({
  type: UPDATE_TA_MARGIN_AMOUNT_ACTION as typeof UPDATE_TA_MARGIN_AMOUNT_ACTION,
  hotelUuid,
  taMarginAmount,
});

export type UpdateIsTAMarginAppliedAction = ReturnType<typeof updateIsTAMarginAppliedAction>;
export const updateIsTAMarginAppliedAction = (hotelUuid: string, value: boolean) => ({
  type: UPDATE_IS_TA_MARGIN_APPLIED_ACTION as typeof UPDATE_IS_TA_MARGIN_APPLIED_ACTION,
  hotelUuid,
  value,
});

export type UpdateBookingGuestInformationAction = ReturnType<typeof updateBookingGuestInformationAction>;
export const updateBookingGuestInformationAction = (bookingGuestInformation: any) => ({
  type: UPDATE_BOOKING_GUEST_INFORMATION_ACTION as typeof UPDATE_BOOKING_GUEST_INFORMATION_ACTION,
  bookingGuestInformation,
});

export type ResetBookingBuilderUiStateAction = ReturnType<typeof resetBookingBuilderUiStateAction>;
export const resetBookingBuilderUiStateAction = () => ({
  type: CLEAR_BOOKING_BUILDER_UI_STATE as typeof CLEAR_BOOKING_BUILDER_UI_STATE,
});

export type UpdateBookingTravelAgentUserIdAction = ReturnType<typeof updateBookingTravelAgentUserIdAction>;
export const updateBookingTravelAgentUserIdAction = (travelAgentUserUuid: string | null = null) => ({
  type: UPDATE_TRAVEL_AGENT_USER_ID as typeof UPDATE_TRAVEL_AGENT_USER_ID,
  travelAgentUserUuid,
});

export type UpdateLodgingRepeatGuestAction = ReturnType<typeof updateLodgingRepeatGuestAction>;
export const updateLodgingRepeatGuestAction = (hotelUuid: string, lodgingIndex: number, checked: boolean) => ({
  type: UPDATE_LODGING_REPEAT_GUEST_ACTION as typeof UPDATE_LODGING_REPEAT_GUEST_ACTION,
  hotelUuid,
  lodgingIndex,
  checked,
});

export type SaveCustomItemAction = ReturnType<typeof saveCustomItemAction>;
export const saveCustomItemAction = (hotelUuid: string) => ({
  type: SAVE_CUSTOM_ITEM as typeof SAVE_CUSTOM_ITEM,
  hotelUuid,
});

export type RemoveCustomItemAction = ReturnType<typeof removeCustomItemAction>;
export const removeCustomItemAction = (index: number, hotelUuid: string) => ({
  type: REMOVE_CUSTOM_ITEM as typeof REMOVE_CUSTOM_ITEM,
  index,
  hotelUuid,
});

export type UpdateAgreeToTermsAction = ReturnType<typeof updateAgreeToTermsAction>;
export const updateAgreeToTermsAction = (value: boolean) => ({
  type: UPDATE_AGREEE_TO_TERMS as typeof UPDATE_AGREEE_TO_TERMS,
  value,
});

export type SetIsPristineAction = ReturnType<typeof setIsPristineAction>;
export const setIsPristineAction = (value: boolean) => ({
  type: SET_IS_PRISTINE as typeof SET_IS_PRISTINE,
  value,
});

export type SetLatestBookingOperationAction = ReturnType<typeof setLatestBookingOperationAction>;
export const setLatestBookingOperationAction = (operation: string) => ({
  type: SET_LATEST_BOOKING_OPERATION as typeof SET_LATEST_BOOKING_OPERATION,
  operation,
});

export type BookingBuilderAction =
  | InitializeBookingBuilderAction
  | CopyBookingBuilderAction
  | CreateStubBookingBuilderAction
  | ClearBookingBuilderAction
  | UpdateTransferAction
  | UpdateBookingSuccessAction
  | UpdateLodgingGuestAgesAction
  | UpdateLodgingDatesAction
  | UpdateLodgingMealPlanAction
  | RemoveLodgingAction
  | UpdateLodgingOccasionsAction
  | AddLodgingAction
  | UpdateGroundServiceAction
  | UpdateSupplementAction
  | UpdateFineAction
  | ForwardsCompatBookingBuilderAction
  | UpdateTAMarginType
  | UpdateTAMarginAmount
  | UpdateIsTAMarginAppliedAction
  | UpdateBookingGuestInformationAction
  | ResetBookingBuilderUiStateAction
  | UpdateBookingTravelAgentUserIdAction
  | UpdateLodgingRepeatGuestAction
  | UpdateAgreeToTermsAction
  | SaveCustomItemAction
  | RemoveCustomItemAction
  | SetIsPristineAction
  | SetLatestBookingOperationAction
  | CustomItemAction;
