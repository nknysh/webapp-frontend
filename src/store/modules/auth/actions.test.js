import { authSignUp, signUp } from './actions';

describe('auth actions', () => {
  describe('authSignUp', () => {
    it('returns action', () => {
      expect(authSignUp({})).toMatchSnapshot();
    });
  });
  describe('signUp', () => {
    it('calls dispatch thunks', () => {
      const dispatch = jest.fn();

      signUp({})(dispatch);

      expect(dispatch).toHaveBeenCalledTimes(2);
    });
  });
});
