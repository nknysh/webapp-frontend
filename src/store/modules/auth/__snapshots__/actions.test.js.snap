// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth actions authSignUp returns action 1`] = `
Object {
  "payload": Object {},
  "type": "AUTH_SIGN_UP",
}
`;
